src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const objsAfterActions = [];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    if (type === 'addProperties') {
+      stateCopy = { ...stateCopy };
+
+      const propForAdd = Object.entries(extraData);
+
+      for (const [key, value] of propForAdd) {
+        stateCopy[key] = value;
+      }
+      objsAfterActions.push(stateCopy);
+    }
+
+    if (type === 'removeProperties') {
+      stateCopy = { ...stateCopy };
+
+      for (const propForRemove of keysToRemove) {
+        if (stateCopy.hasOwnProperty(propForRemove)) {
+          delete stateCopy[propForRemove];
+        }
+      }
+      objsAfterActions.push(stateCopy);
```

Comment:
You do this action in each case. So just move it to the end of the function to not repeat yourself